packages/sdk/src/wasm/identity.ts

```diff
@@ -10,47 +10,162 @@ import type {
 } from '../types.js';
 import type { IdProofInput, IdProofOutput } from '../id/index.js';
 
-export type IdentityRequestInput = {
+interface IdentityRequestInputCommon {
     ipInfo: IpInfo;
     globalContext: CryptographicParameters;
     arsInfos: Record<string, ArInfo>;
+    arThreshold: number;
+}
+
+export type IdentityRequestInput = IdentityRequestInputCommon & {
     seed: string;
     net: Network;
     identityIndex: number;
-    arThreshold: number;
 };
 
+type IdentityRequestInputInternal = {
+    common: IdentityRequestInputCommon;
+    net: Network;
+    seedAsHex: string;
+    identityIndex: number;
+};
+
+export type IdentityRequestWithKeysInput = IdentityRequestInputCommon & {
+    prfKey: string;
+    idCredSec: string;
+    blindingRandomness: string;
+};
+
+type IdentityRequestWithKeysInputInternal = {
+    common: IdentityRequestInputCommon;
+    prfKey: string;
+    idCredSec: string;
+    blindingRandomness: string;
+};
+
+/**
+ * Creates a V1 identity request by providing the secret keys directly.
+ * This allows for the generation of the keys separately from creating
+ * the request.
+ */
+export function createIdentityRequestWithKeys(
+    input: IdentityRequestWithKeysInput
+): Versioned<IdObjectRequestV1> {
+    const { prfKey, idCredSec, blindingRandomness, ...common } = input;
+    const internalInput: IdentityRequestWithKeysInputInternal = {
+        common,
+        prfKey,
+        idCredSec,
+        blindingRandomness,
+    };
+
+    const rawRequest = wasm.createIdRequestWithKeysV1(
+        JSON.stringify(internalInput)
+    );
+    try {
+        return JSON.parse(rawRequest).idObjectRequest;
+    } catch (e) {
+        throw new Error(rawRequest);
```

Comment:
Maybe I'm missing something here, but why you know why we throw an error with the response from the wasm function? I see that this is what we do everywhere, but it seems off to me (unless we use the error case for something somewhere?).

If we do use the error case, perhaps we should add a `@throws` declaration to the jsdoc describing the use case

Reply:
I'm fairly sure the error case of the rust `Result` is translated to javascript `throw`, with the concrete error type (`JsError`) returned in the `Result` is translated to a javascript `Error` instance, so in that case this only checks that the successful wasm invocation is parsable by `JSON.parse`.

Reply:
I repeated Mr. Hjorts comment :smile: 

If what you are saying is the case, then the error gotten directly from JSON.parse is probably better to just get directly. Or perhaps add `rawRequest` to the error message gotten there as some more context helps debugging the JSON errors. It usually gives us something like:
```
VM318:1 Uncaught SyntaxError: Unexpected token 'F', "Failure" is not valid JSON
    at JSON.parse (<anonymous>)
    at <anonymous>:1:6
```
and without the given JSON it can sometimes be hard to find exactly where the issue lies. Especially when the JSON is huge.

Can we agree to keep it as is within this PR, and if we want to change it in general, then it should be done separately?